app/src/main/java/com/example/mytestproject/ui/weatherData/WeatherDataFragment.kt

```diff
@@ -39,12 +41,35 @@ class WeatherDataFragment : Fragment() {
         animationDrawable.start()
 
         setupWeatherData()
+
+        btnRetry.setOnClickListener {
+            weatherDataViewModel.getWeather()
```

Comment:
Usually onClick listeners don't call specific methods of model, because it looks like an implementing of logic in UI. Fragment knows what to do, but it shouldn't.
The better solution is a creating in ViewModel a method with name onRetry (or similar). Fragment calls this method when Retry button is clicked. And model decides what to do exactly.
So, fragment just says that the button was clicked, but doesn't ask to request weather.